### PR TITLE
Rivescript integration

### PR DESCRIPTION
**Pull request by dcsan**
this is a placeholder PR with an example rivescript integration. 
The PR is a bit messy but I wanted to get something in time for the slackdevs meetup tonite, I'll cleanup with input from others too.

you probably don't want to merge this because of some mods to your error handling etc

https://github.com/howdyai/botkit/compare/master...dcsan:rivescript-integration?expand=1#diff-ffa8ec227ea6e9babdfc11d5a71ee29dL87

already updated. if anyone is interested better to clone from
https://github.com/dcsan/botkit

update1 - I chatted with Ben about this and he was thinking we should maybe do this as a middleware package instead? It may introduce too much bloat to the core botkit, so perhaps I'll wait until it's clear which way you'll go?

update2 - i made fixes wrt most of the requests below, up to you if you want to merge.

### CODE REVIEW

File: bin/ngrok [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+ngrok http 3030
```
Comment (dcsan) [inline bin/ngrok]: you probably dont want these, but i like to keep a few helpers around

File: bin/ribot [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+source bin/local.env
```
Comment (dcsan) [inline bin/ribot]: this assumes you keep your token etc in a *.env file

File: bot.js [hunk 3]
```diff
@@ -167,6 +167,7 @@ controller.hears(['uptime','identify yourself','who are you','what is your name'
 
 });
 
+
```
Comment (dcsan) [inline bot.js]: oops?

File: lib/SlackBot.js [hunk 4]
```diff
@@ -81,24 +81,26 @@ function Slackbot(configuration) {
...
-                        message.type = 'slash_command';
-                        // HEY THERE
-                        // Slash commands can actually just send back a response
-                        // and have it displayed privately. That means
-                        // the callback needs access to the res object
-                        // to send an optional response.
+                    }
+                    // for now just continue
+                    // } else {
+                    message.type = 'slash_command';
```
Comment (dcsan) [inline lib/SlackBot.js]: i'll remove this file/mods from the final PR, unless things change at your end wrt
https://github.com/howdyai/botkit/pull/40

File: examples/ribot/BotLoader.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const RiveScript = require("riveScript");
```
Comment (anonrig) [inline examples/ribot/BotLoader.js]: We're not using ES6.

File: examples/ribot/BotLoader.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+const path = require("path");
+const BotLoader = {};
+
+// Load a directory full of RiveScript documents (.rive files)
+const brainsDir = path.join(__dirname, "scripts");
+
+
+BotLoader.reset = function() {
+    console.log("loading rivescripts brain");
```
Comment (anonrig) [inline examples/ribot/BotLoader.js]: Please don't use `console.log` instead follow our `logger`

File: examples/ribot/BotLoader.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    BotLoader.brain = new RiveScript();
+    BotLoader.brain.loadDirectory(brainsDir, loading_done, loading_error);
+}
+
+// Load an individual file.
```
Comment (anonrig) [inline examples/ribot/BotLoader.js]: We follow JSDoc for documentation. Please read `CONTRIBUTING.md`

File: examples/ribot/shell.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env node
+
+/******************************************************************************
```
Comment (anonrig) [inline examples/ribot/shell.js]: Please use JSDOC.

File: lib/SlackBot.js [hunk 9]
```diff
@@ -81,24 +81,26 @@ function Slackbot(configuration) {
...
-                        bot.team_info = team;
-                        bot.res = res;
+                    bot.team_info = team;
+                    bot.res = res;
 
-                        slack_botkit.receiveMessage(bot, message);
+                    slack_botkit.receiveMessage(bot, message);
 
-                    }
+                    // }
```
Comment (anonrig) [inline lib/SlackBot.js]: Please remove commented lines.

**Comment by anonrig** [comment]
Please follow `CONTRIBUTING.md` for commit messages.

File: examples/ribot/BotLoader.js (hunk 7 above)
Comment (dcsan) [inline examples/ribot/BotLoader.js]: @anonrig 
do you have an .eslintrc ? your current code seems pretty varied.

Comment (anonrig) [inline examples/ribot/BotLoader.js]: We have `.jscsrc`, but it will be better to have both

Comment (dcsan) [inline examples/ribot/BotLoader.js]: I can throw an eslint into the tree for the stuff I was working on but my preferred settings don't seem to match yours.

**Comment by RafaelCosman** [comment]
@dcsan Thanks for making this!

File: examples/ribot/shell.js (hunk 8 above)
Comment (dcsan) [inline examples/ribot/shell.js]: this is how the original rivescript is authored, don't want to change that

File: lib/SlackBot.js (hunk 9 above)
Comment (dcsan) [inline lib/SlackBot.js]: they can't be removed - waiting for an upstream fix here.

**Comment by anonrig** [comment]
Please fix the conflicts and re-open the pull request.

**Pull request closed without merging**